distributed/tests/test_profile.py

```diff
@@ -211,27 +211,29 @@ def stop():
 def test_watch_requires_lock_to_run():
     start = time()
 
-    def stop_lock():
-        return time() > start + 0.600
+    release_lock = False
 
-    def stop_profile():
+    def stop_blocking():
+        return release_lock
+
+    def stop_profiling():
         return time() > start + 0.500
 
-    def hold_lock(stop):
+    def block_lock(stop):
         with lock:
             while not stop():
                 sleep(0.1)
 
     start_threads = threading.active_count()
 
-    # Hog the lock over the entire duration of watch
+    # Block the lock over the entire duration of watch
     thread = threading.Thread(
-        target=hold_lock, name="Hold Lock", kwargs={"stop": stop_lock}
+        target=block_lock, name="Block Lock", kwargs={"stop": stop_blocking}
     )
     thread.daemon = True
     thread.start()
 
-    log = watch(interval="10ms", cycle="50ms", stop=stop_profile)
+    log = watch(interval="10ms", cycle="50ms", stop=stop_profiling)
 
     start = time()  # wait until thread starts up
```

Comment:
how about using an event to wait for the profiler thread to start?

```python
stop_profiling_called = threading.Event()

def stop_profiling():
    stop_profiling_called.set()
    return return time() > start + 0.500

stop_profiling_called.wait(2)
```

Reply:
Done, adjusted the structure to be more event-based.